avldb/src/main/scala/scorex/db/LDBVersionedStore.scala

```diff
@@ -407,4 +407,26 @@ class LDBVersionedStore(protected val dir: File, val initialKeepVersions: Int) e
   def rollbackVersions(): Iterable[VersionID] = {
     versions.reverse
   }
+
+  /**
+    * Take database snapshot, process it, and then close the snapshot.
+    * Could be useful when it is needed to process current state of database without blocking database (and so threads
+    * possibly working with it).
+    *
+    * @param logic - processing logic which is getting access to `get` function to read from database snapshot
+    */
+  def processSnapshot[T](logic: {def get(key: Array[Byte]): Array[Byte]} => T): T = {
```

Comment:
Why use such type of `logic`. I suggest to introduce Loader abstract class inherited from Function1 and use it here.

Reply:
I started with something like 

```
trait ReadInterface{
  def get(key: Array[Byte]
}
```

and def processSnapshot[T](logic: ReadInterface => T): T

But `ReadInterface` looks artificial, anon function complex, the structural type does it job pretty well imho

Reply:
Ok, introduced read interface again

Reply:
Looks good.